Cabal/src/Distribution/PackageDescription/Check.hs

```diff
@@ -1385,7 +1385,7 @@ checkGhcOptions fieldName getOptions pkg =
   , checkFlags ["-prof"] $
       PackageBuildWarning (OptProf fieldName)
 
-  , checkFlags ["-o"] $
+  , unlessScript . checkFlags ["-o"] $
```

Comment:
Why are you modifying `check` behaviour? What has this to do with the PR? Specifically, is this necessary for 2004ca697233ed4faa600699e0169931bf44ff53 (`Use the actual script name in the executable name`)?

Reply:
That is correct. I can either add a more explicit comment or I can back out the change if people feel it's not appropriate.

Reply:
Thanks, sorry for late reply. Can you explain to me why it is needed? How does it break testsuite?

Reply:
I will mark this as "resolved" since there is an understandable willingness to get this patch promptly into `master`, but I still would like an explanation when you have time!

Reply:
@ffaf1 Maybe, create an issue and tag the author? I fully share your concern by the way.